strawberry/extensions/tracing/sentry.py

```diff
@@ -0,0 +1,125 @@
+import hashlib
+from inspect import isawaitable
+from typing import Optional
+
+from sentry_sdk import configure_scope, start_span
+
+from strawberry.extensions import Extension
+from strawberry.extensions.tracing.utils import should_skip_tracing
+from strawberry.types.execution import ExecutionContext
+from strawberry.utils.cached_property import cached_property
+
+
+class SentryTracingExtension(Extension):
+    def __init__(
+        self,
+        *,
+        execution_context: Optional[ExecutionContext] = None,
+    ):
+        if execution_context:
+            self.execution_context = execution_context
+
+    @cached_property
+    def _resource_name(self):
+        assert self.execution_context.query
+
+        query_hash = self.hash_query(self.execution_context.query)
+
+        if self.execution_context.operation_name:
+            return f"{self.execution_context.operation_name}:{query_hash}"
+
+        return query_hash
+
+    def hash_query(self, query: str):
+        return hashlib.md5(query.encode("utf-8")).hexdigest()
+
+    def on_request_start(self) -> None:
+        self._operation_name = self.execution_context.operation_name
+        name = f"{self._operation_name}" if self._operation_name else "Anonymous Query"
+
+        with configure_scope() as scope:
+            if scope.span:
+                self.gql_span = scope.span.start_child(
+                    op="gql",
+                    description=name,
+                )
+            else:
+                self.gql_span = start_span(
+                    op="gql",
+                )
+
+        operation_type = "query"
+
+        assert self.execution_context.query
+
+        if self.execution_context.query.strip().startswith("mutation"):
+            operation_type = "mutation"
+        if self.execution_context.query.strip().startswith("subscription"):
+            operation_type = "subscription"
+
+        self.gql_span.set_tag("graphql.operation_type", operation_type)
+        self.gql_span.set_tag("graphql.resource_name", self._resource_name)
+        self.gql_span.set_data("graphql.query", self.execution_context.query)
+
+    def on_request_end(self) -> None:
+        self.gql_span.finish()
+
+    def on_validation_start(self):
+        self.validation_span = self.gql_span.start_child(
+            op="validation", description="Validation"
+        )
+
+    def on_validation_end(self):
+        self.validation_span.finish()
+
+    def on_parsing_start(self):
+        self.parsing_span = self.gql_span.start_child(
+            op="parsing", description="Parsing"
+        )
+
+    def on_parsing_end(self):
+        self.parsing_span.finish()
+
```

Comment:
Hi @patrick91, what do you think adding

```
    def on_execute(self):
        with start_transaction(
                op=self._operation_type,
                name=self._operation_name,
        ):
            yield
```

With this we can also have the option to track the transaction in sentry 😊

Reply:
@Ronjea I'll do this once I get back to this PR, thanks for the suggestion!

Reply:
My pleasure ☺️

Reply:
@Ronjea can you take a look at this PR? 

I'll merge this soon, but happy to update the extension with your suggestions later on